src/cryptonote_core/miner.cpp

```diff
@@ -365,6 +405,19 @@ namespace cryptonote
         misc_utils::sleep_no_w(100);
         continue;
       }
+      else if( m_is_background_mining_enabled )
+      {
+        misc_utils::sleep_no_w(m_miner_extra_sleep);
+        while( !m_is_background_mining_started )
+        {
+          MGINFO("background mining is enabled, but not started, waiting until start triggers");
+          boost::unique_lock<boost::mutex> started_lock( m_is_background_mining_started_mutex );        
+          m_is_background_mining_started_cond.wait( started_lock );
+          if( m_stop ) break;
+        }
+        
+        if( m_stop ) continue;         
```

Comment:
If **m_is_background_mining_started_cond** is notified by the stop() call, control breaks from the **while( !m_is_background_mining_started )** loop. I could let the function run through one more time, but I figured it's just more responsive to stop on the spot rather then calculate a few more hashes.